docs/table-migration.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Table Migration"
+url: table-migration
+weight: 1300
+menu: main
+---
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+# Table Migration
+Apache Iceberg supports converting existing tables in other formats to Iceberg tables. This section introduces the general concept of table migration, its approaches, and existing implementations in Iceberg.
+
+## Migration Approaches
+There are two main approaches to perform table migration: CTAS (Create Table As Select) and in-place migration.
+
+### Create-Table-As-Select Migration
+CTAS migration involves creating a new Iceberg table and copying data from the existing table to the new one. This method is preferred when you want to completely cut ties with your old table, ensuring the new table is independent and fully managed by Iceberg.
```

Comment:
I think we can skip talking about "full data migration case" in too much depth, and just refer to links of the write operations of specific engines for CTAS, INSERT, and change data capture. We don't even need a section for it, we just need to say in the intro that in this doc we will describe more about "In-place metadata migration"